db/migrate/20170720125756_create_product_nature_variant_valuings.rb

```diff
@@ -0,0 +1,10 @@
+class CreateProductNatureVariantValuings < ActiveRecord::Migration
+  def change
+    create_table :product_nature_variant_valuings do |t|
+      t.decimal :average_cost_amount, precision: 19, scale: 4, null: false
+      t.decimal :amount, precision: 19, scale: 4,              null: false
+      t.references :variant,                                   null: false, index: true
+      t.timestamps                                             null: false
```

Comment:
Use `t.stamps` instead of `t.timestamps`. It's a homemade method to add more tracking columns.